llvm/include/llvm/Support/Compiler.h

```diff
@@ -569,4 +569,12 @@ void AnnotateIgnoreWritesEnd(const char *file, int line);
 #define LLVM_NO_PROFILE_INSTRUMENT_FUNCTION
 #endif
 
+/// \macro LLVM_PREFERRED_TYPE
+/// Adjust type of bit-field in debug info.
+#if __has_attribute(preferred_type)
```

Comment:
```suggestion
#if LLVM_HAS_CPP_ATTRIBUTE(clang::preferred_type)
```
or leave the condition as-is and change the expansion to using `__attribute__(())` instead. Just making sure they both match up. (If we're using this from C interfaces, then we probably want to use the GNU-style spelling.)